### PR TITLE
adds support for list of monitors in detector

### PR DESCRIPTION
**Pull request by eirsep**
Signed-off-by: Surya Sashank Nistala <snistala@amazon.com>

### CODE REVIEW

File: src/main/java/org/opensearch/securityanalytics/model/Detector.java [hunk 1]
```diff
@@ -355,20 +361,20 @@ public static Detector parse(XContentParser xcp, String id, Long version) throws
...
+                version,
+                Objects.requireNonNull(name, "Detector name is null"),
+                enabled,
+                Objects.requireNonNull(schedule, "Detector schedule is null"),
+                lastUpdateTime != null ? lastUpdateTime : Instant.now(),
+                enabledTime,
+                DetectorType.valueOf(detectorType.toUpperCase(Locale.ROOT)),
+                user,
+                inputs,
+                monitorIds,
```
Comment (getsaurabh02) [inline src/main/java/org/opensearch/securityanalytics/model/Detector.java:374]: Do we also need Monitor Versions here?

File: src/test/java/org/opensearch/securityanalytics/resthandler/DetectorRestApiIT.java [hunk 2]
```diff
@@ -50,7 +50,7 @@ public void testCreatingADetector() throws IOException {
         Assert.assertFalse(((Map<String, Object>) responseBody.get("detector")).containsKey("findings_index"));
         Assert.assertFalse(((Map<String, Object>) responseBody.get("detector")).containsKey("alert_index"));
 
-        String monitorId = ((Map<String, Object>) responseBody.get("detector")).get("monitor_id").toString();
+        String monitorId = ((List<String>) ((Map<String, Object>) responseBody.get("detector")).get("monitor_id")).get(0);
```
Comment (getsaurabh02) [inline src/test/java/org/opensearch/securityanalytics/resthandler/DetectorRestApiIT.java:53]: Lets create a test which indexes multiple monitors and verify detector response.

File: src/main/java/org/opensearch/securityanalytics/model/Detector.java (hunk 1 above)
Comment (eirsep) [reply src/main/java/org/opensearch/securityanalytics/model/Detector.java:374]: We won't need monitor versions as we can fetch monitors just using id.

File: src/test/java/org/opensearch/securityanalytics/resthandler/DetectorRestApiIT.java (hunk 2 above)
Comment (eirsep) [reply src/test/java/org/opensearch/securityanalytics/resthandler/DetectorRestApiIT.java:53]: added response parsing tests
Created issue #51 to track indexing multiple monitors.

Currently only added support for handling list of monitors in detector model and response.

**Review by sbcd90 — APPROVED** [review]

**Review by getsaurabh02 — APPROVED** [review]

**Pull request merged**